### DISCUSSION TITLE
How about add a static transaction method to model?

### DISCUSSION THREAD

**Discussion by chuoke** [laravel/framework#49136] [Ideas]
I didn't do much research. That's how I'm using it at the moment:

```php
// Option 1 mostly
DB::transaction(function () {
   // code...
});

// Option 2
$userModel = new UserModel();
$userModel->getConnection()->transaction(function () {
   // code...
});
```

But this is not very convenient, especially if you do not use the default connection.

```php
DB::connection('name')->transaction(function () {
   // code...
});
```
I often forget the `connection` method, or need to find the corresponding name, if the name has been modified, and a large number of names used is prone to omission.
But if you already know which models to manipulate, why not just use the models to handle them?
```php
  UserModel::transaction(function () {
     // code
  });
```

What do you think?

**Comment by crynobone** [comment]
You can already do:

```php
$userModel->saveOrFail();
```

**Reply by chuoke** [reply]
Not this. I may be vague, but there are many store actions in a transaction.

**Comment by ghost** [comment]
You can create it for your models. A function in a base model or in a trait

    /**
    * @throws \Throwable
    */
    public static transaction(callable $f): void
    {
        $model = new self();
        $model->getConnection()->transaction($f);
    }

**Reply by chuoke** [reply]
yeah, I can do this. 
But I would like to know if this is a requirement for many people, and if so, if this shortcut can be added to the framework.
